### PR TITLE
[new module] azure_rm_sqlfirewallrule

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Adding support for azure_rm_sqlfirewallrule

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME

azure_rm_sqlfirewallrule

##### ANSIBLE VERSION

 2.4

### CODE REVIEW

**Comment by zikalino** [comment]
ready_for_review

**Comment by george-wilson-iproov** [comment]
Tested succesfully

File: lib/ansible/modules/cloud/azure/azure_rm_sqlfirewallrule.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+                                                   supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()):
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+
+        old_response = None
```
Comment (george-wilson-iproov) [inline lib/ansible/modules/cloud/azure/azure_rm_sqlfirewallrule.py]: You're overwriting this immediately on line 150

File: lib/ansible/modules/cloud/azure/azure_rm_sqlfirewallrule.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_sqlfirewallrule
+version_added: "2.5"
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqlfirewallrule.py]: 2.7

File: lib/ansible/modules/cloud/azure/azure_rm_sqlfirewallrule.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        self.to_do = Actions.NoAction
+
+        super(AzureRMFirewallRules, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                   supports_check_mode=True,
+                                                   supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()):
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqlfirewallrule.py:144]: + ['tags']?

File: lib/ansible/modules/cloud/azure/azure_rm_sqlfirewallrule.py [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+
+        old_response = None
+        response = None
+
+        self.mgmt_client = self.get_mgmt_svc_client(SqlManagementClient,
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_sqlfirewallrule.py]: add this to azure_rm_common.py

**Comment by yungezz** [comment]
@zikalino  pls resolve comments

**Comment by Fred-sun** [comment]
@zikalino Could you please help update the issue? Thanks!

**Comment by Fred-sun** [comment]
@zikalino could you please update the comment? Thanks!

**Comment by Fred-sun** [comment]
@zikalino Thanks for you fix sample, does it ready for review again?

File: lib/ansible/modules/cloud/azure/azure_rm_sqlfirewallrule.py (hunk 3 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_sqlfirewallrule.py:144]: firewall rule has no tags

**Comment by Fred-sun** [comment]
@zikalino This PR exist conflict, Could you help to resolve and push it to ready_for_review? thanks!

**Comment by Fred-sun** [comment]
@zikalino Thanks for your update the change. Could you help confirm this PR is ready for review or not? Thanks!

**Comment by zikalino** [comment]
@Fred-sun yes, it's ready, I just rebased it

**Comment by Fred-sun** [comment]
@zikalino Thank you for these change, I will push review. Thanks!

**Comment by Fred-sun** [comment]
@yungezz @george-wilson-iproov This PR has ready for review, Could you help to review and push to merge? Thanks!

**Comment by Fred-sun** [comment]
@yungezz @george-wilson-iproov This PR has ready for review, Could you help to review and push to next step forward? Thanks!

**Comment by Fred-sun** [comment]
@zikalino This PR exist conflict, Could you help to resolve and push it to review? Thanks!

**Comment by nitzmahone** [comment]
merged via 7d81de251640e5c3d7a24d75245bbe875004686b

**Pull request closed without merging**